base/src/ArchiveSpaceManager.cpp

```diff
@@ -0,0 +1,180 @@
+#include <stdafx.h>
+#include<cstdlib>
+#include <boost/filesystem.hpp>
+#include "Module.h"
+#include "ArchiveSpaceManager.h"
+
+class ArchiveSpaceManager::Detail
+{
+public:
+    Detail(ArchiveSpaceManagerProps& _props) : mProps(_props)
+    {
+    }
+
+    ~Detail()
+    {
+    }
+
+    void setProps(ArchiveSpaceManagerProps _props)
+    {
+        mProps = _props;
+    }
+  
+    uint32_t estimateDirectorySize(boost::filesystem::path _dir)
+    {
+        uint32_t dirSize = 0;
+        int sample = 0;
+        int inCount = 0;
+        int countFreq = 0;
+        uint32_t tempSize = 0;
+       
+        for (const auto& entry : boost::filesystem::recursive_directory_iterator(_dir))
+        {
+            if ((boost::filesystem::is_regular_file(entry)))
+            {
+                if (!(countFreq % mProps.samplingFreq))
+                {
+                    sample = (rand() % mProps.samplingFreq) + 1;
+                    inCount = 0;
+                }
+                if (inCount == sample)
+                {
+                    tempSize = boost::filesystem::file_size(entry);
+                    dirSize += tempSize * mProps.samplingFreq;
```

Comment:
what if the total number of files is not a multipe of mProps.samplingFreq? e.g. what if you have 46-59 files in an hour ? the last batch now has 1-14 files while you are multiplying by 15

Reply:
The no of files in last batch are multiplied with tempSize and added to dirSize after for loop.